CoreUtilities/src/au/gov/asd/tac/constellation/utilities/text/SpellChecker.java

```diff
@@ -0,0 +1,158 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.utilities.text;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
+import javafx.stage.Popup;
+import org.apache.commons.lang3.StringUtils;
+import org.languagetool.JLanguageTool;
+import org.languagetool.MultiThreadedJLanguageTool;
+import org.languagetool.language.AustralianEnglish;
+import org.languagetool.rules.RuleMatch;
+
+/**
+ * Handles the SpellChecking functions of SpellCheckingTextArea. SpellChecker
+ * evaluates incorrect words/phrases and pops up the suggestions when the user
+ * prompts
+ *
+ * @author Auriga2
+ */
+public final class SpellChecker {
+
+    private final SpellCheckingTextArea textArea;
+    private static final List<String> misspells = new ArrayList<>();
+    private List<RuleMatch> matches = new ArrayList<>();
+    private int currentIndex;       // index for current position in misspells list
+    private final ListView<String> suggestions = new ListView<>(FXCollections.observableArrayList());
+    private String misspelledTextUnderCursor;
+    private final JLanguageTool langTool;
+    private Popup popup;
+    private boolean turnOffSpellChecking = false;
+
+    public SpellChecker(final SpellCheckingTextArea spellCheckingTextArea) {
+        textArea = spellCheckingTextArea;
+        langTool = new MultiThreadedJLanguageTool(new AustralianEnglish()); //Can pass in the language when supporting multiple languages
+        initialize();
+    }
+
+    /**
+     * set a listener on the list views selection model to get the text of the
+     * selected row.
+     */
+    private void initialize() {
+        popup = new Popup();
+        suggestions.getSelectionModel().selectedItemProperty().addListener((obs, oldValue, newValue) -> {
+            if (newValue != null) {
+                textArea.replaceText​(textArea.getText().replace(misspelledTextUnderCursor, newValue));
+            }
+            popup.hide();
+        });
+    }
+
+    /**
+     * Check Spelling of the entire text. This will ensure scenarios like
+     * duplicate words, grammar mistakes etc. are triggered
+     */
+    public void checkSpelling() {
+        textArea.clearStyles();
+        misspells.clear();
+        if (!turnOffSpellChecking && !StringUtils.isBlank(textArea.getText())) {
+            try {
+                matches = langTool.check(textArea.getText());
+            } catch (IOException ex) {
+                Logger.getLogger(SpellChecker.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
can we make the logger a constant for the class (similar to what has been done on other classes in the code. Also if the message could instead be `ex.getLocalisedMessage()` rather than null